python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -404,7 +404,9 @@ def _map_task(
     stats = BlockExecStats.builder()
     for b_out in map_transformer.apply_transform(iter(blocks), ctx):
         # TODO(Clark): Add input file propagation from input blocks.
-        m_out = BlockAccessor.for_block(b_out).get_metadata([], None)
+        m_out = BlockAccessor.for_block(b_out).get_metadata(
+            [], None, location=ray.get_runtime_context().get_node_id()
```

Comment:
this only set location for the map op. can we move this to `get_metadata`?

Reply:
I think sometimes get_metadata is called on the driver and not the task worker, so `get_node_id()` won't return the right value.

Reply:
I guess we need to make sure it is set on all task outputs. Do you know of an easy way to do that?

Reply:
Currently we don't have a unified place to outputting blocks. I plan to refactor the code and add that in 2.8.
I checked the existing get_metadata call sites. It seems that all of them are on the task workers. Do you have an exceptional example? I guess it's fine to exclude a few of them.